pkg/apis/v1/ec2nodeclass_status.go

```diff
@@ -17,14 +17,16 @@ package v1
 import (
 	"github.com/awslabs/operatorpkg/status"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
-	ConditionTypeSubnetsReady         = "SubnetsReady"
-	ConditionTypeSecurityGroupsReady  = "SecurityGroupsReady"
-	ConditionTypeAMIsReady            = "AMIsReady"
-	ConditionTypeInstanceProfileReady = "InstanceProfileReady"
-	ConditionTypeValidationSucceeded  = "ValidationSucceeded"
+	ConditionTypeSubnetsReady              = "SubnetsReady"
+	ConditionTypeSecurityGroupsReady       = "SecurityGroupsReady"
+	ConditionTypeAMIsReady                 = "AMIsReady"
+	ConditionTypeInstanceProfileReady      = "InstanceProfileReady"
+	ConditionTypeCapacityReservationsReady = "CapacityReservationsReady"
```

Comment:
It seems a little odd to have the CapacityReservation condition for NodeClasses that may not be selecting on terms for capacityReservations at all -- this is probably the best way to signal that we've resolved the latest terms though -- at least gives us some indication when the spec changes that we need to wait until things update

Reply:
Discussed offline, agreed to keep the status condition for the reasons outlined here. We will drop this condition when the feature flag is disabled though.

Reply:
Noting that it seems like we still need to implement the gating behavior